crates/sui-bridge-indexer/src/eth_bridge_indexer.rs

```diff
@@ -183,6 +183,25 @@ impl Datasource<RawEthData> for EthSyncDatasource {
         let handle = spawn_monitored_task!(async move {
             let mut cached_blocks: HashMap<u64, Block<H256>> = HashMap::new();
 
+            let mut last_finalized_block = retry_with_max_elapsed_time!(
+                client.get_last_finalized_block_id(),
+                Duration::from_secs(30000)
+            )
+            .expect("Unable to get latest finalized block from provider")
+            .expect("Failed to unwrap the inner result");
```

Comment:
rather than having multiple tasks query this, we should have one dedicated thread check this and update the watch channel, just like what we do for EthSyncer